src/main/java/com/zepben/testutils/exception/ExpectException.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package com.zepben.testutils.exception
+
+import kotlin.Throws
+
+class ExpectException private constructor(val func: RunWithException) {
+    fun interface RunWithException {
+        @Throws(Throwable::class)
+        fun run()
+    }
+
+    @Throws(ExpectExceptionError::class)
+    inline fun <reified T: Throwable> toThrow(): ExceptionMatcher<T> {
+        try {
+            func.run()
+        } catch (e: Throwable) {
+            if (e is T) return ExceptionMatcher(e)
+            throw ExpectExceptionError(T::class.simpleName, e.javaClass.simpleName, e)
+        }
+        throw ExpectExceptionError(T::class.simpleName, "")
+    }
+
+    /**
+     * Java-interoperable version of toThrow()
+     */
+    @Throws(ExpectExceptionError::class)
+    fun <T: Throwable> toThrow(clazz: Class<T>): ExceptionMatcher<T> {
```

Comment:
No covereage of this function. This is what was meant by my previous comment about `.toThrow()` from java, which should probably have read `.toThrow(<CLASS>)` as it was not intended to try and call the non-generic version, but highlight the interop was missing and you can't call the inline version from java.